ignite/contrib/handlers/base_logger.py

```diff
@@ -103,7 +105,35 @@ def _setup_output_metrics(self, engine: Engine) -> Dict[str, Any]:
                 output_dict = {"output": output_dict}
 
             metrics.update({name: value for name, value in output_dict.items()})
-        return metrics
+
+        metrics_dict = {}  # type: Dict[Any, Union[str, float, numbers.Number]]
+
+        for name, value in metrics.items():
+            if isinstance(value, numbers.Number):
+                if key_tuple:
+                    metrics_dict[(self.tag, name)] = value
+                else:
+                    metrics_dict[f"{self.tag}/{name}"] = value
```

Comment:
Let's define a key transforming function once, like here:
```python
key_tuple = False

if key_tuple:
    key_tf = lambda tag, name, *args: (tag, name) + args
else:
    key_tf = lambda tag, name, *args: "/".join((tag, name) + args)
    
key_tf("tag", "name"), key_tf("tag", "name", "1")
> ('tag/name', 'tag/name/1')
> (('tag', 'name'), ('tag', 'name', '1'))
```
and only use it without multiple if-else.

Reply:
Yup, thats much better. I have changed